src/main/scala/rps/Main.scala

```diff
@@ -1,54 +1,46 @@
-
-    One convention is to name the package the same as the
-    directory containing the Scala file.
-    However, Scala is agnostic to file layout.
- */
 package rps
 
-object Main {
-  def main(args: Array[String]): Unit = {
-    println("""
-#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-%&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&
-%&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&
-%&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&
-%&&&&&&&&&&&&&&&&&&&&&&&&&&&&********%&&&&&&&&&&&&&&&&&&&&&
-%&&&&&&&&&&&&&&&&&&&&&&&&&&********%&&&&&&&&&&&&&&&&&&&&&&&
-%&&&&&&&&&&&&&&&&&&&&&&&&/*********/&&&&&&&&&&&&&&&&&&&&&&&
-%&&&&&&&&&&&&&&&&&&&&&&***************&&&&&&&&&&&&&&&&&&&&&
-#&&&&&&&&&&&&&&&&&&&&********#&%********&&&&&&&&&&&&&&&&&&&
-#&&&&&&&&&&&&&&&&&&/*******#&&&&&#*******/&&&&&&&&&&&&&&&&&
-#&&&&&&&&&&&&&&&&(*******(&&&&%&&&&#*******/&&&&&&&&&&&&&&&
-#&&&&&&&&&&&&&&/*******#&&&&#***(&&&&%********&&&&&&&&&&&&&
-#&&&&&&&&&&&&(*******#&&&&#*******(&&&&%*******/&&&&&&&&&&&
-#&&&&/#&&&&/*******#&&&&/************&&&&%********&&&&&&&&&
-#%%%%///#/*******#%%%%#***************(&%%%%********&%%%%%%
-#%%%%///*******#%%%%#*******************(%%%%%*******/%%%%%
-#%%%%********#%%%%/************************%%%%%********%%%
-#%%%%%/*******#%%%%#*********%%%*********(%%%%#********/%%%
-#%%%%%%%/*******#%%%%(*****%%%%%%%*****/%%%%#********///%%%
-#%%%%%%%%%(*******#%%%%#/%%%%%%%%%%%/(%%%%#*******/%%%//%%%
-#%%%%%%%%%%%(*******#%%%%%%%%%%%%%%%%%%%#********%%%%%%%%%%
-#%%%%%%%%%%%%%/******%%%%%%%%%%%%%%%%%%%*******%%%%%%%%%%%%
-#%%%%%%%%%%%%%%%(*/%%%%%%%%%%%%%%%%%%%%%%%/*/%%%%%%%%%%%%%%
-#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.server.Directives._
+import akka.stream.ActorMaterializer
+import scala.io.StdIn
 
-""")
-    println("Let's play!\n")
-    Game.play()
-  }
-  /* Alternatively we could do:
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
+import io.circe.generic.auto._
+import io.buildo.enumero.circe._
+
+object WebServerRPS {
+  def main(args: Array[String]) {
 
-    ```scala
-        object Main extends App {
-        println("Wanna play?)
+    implicit val system = ActorSystem("my-system")
+    implicit val materializer = ActorMaterializer()
+    // needed for the future flatMap/onComplete in the end
+    implicit val executionContext = system.dispatcher
+
+    val router =
+    pathPrefix("rps") {
+      path("play") {
+        post {
+          entity(as[RPSRequest]) { request => /// will unmarshal JSON to RPSRequest
+              val result = Game.play(request.userMove)
+              result match {
+                case None => complete("")
```

Comment:
This is ok for now, but it's not pretty ;)
We'll see how to return a meaningful error instead.